osu.Game/Screens/Play/GameplayBeatmap.cs

```diff
@@ -16,9 +16,14 @@ public class GameplayBeatmap : Component, IBeatmap
     {
         public readonly IBeatmap PlayableBeatmap;
 
-        public GameplayBeatmap(IBeatmap playableBeatmap)
+        private readonly GameplayClockContainer gameplayClockContainer;
+
+        public GameplayBeatmap(IBeatmap playableBeatmap, GameplayClockContainer gameplayClockContainer = null)
         {
             PlayableBeatmap = playableBeatmap;
+
+            if (gameplayClockContainer != null)
+                this.gameplayClockContainer = gameplayClockContainer;
```

Comment:
I am not sure `GameplayBeatmap` should be holding a reference to the entire `GameplayClockContainer`. I'd much prefer `GameplayBeatmap` only binding to `GameplayClockContainer.UserPlaybackRate` itself instead.

Reply:
Something like this
`public BindableNumber<double> UserPlaybackRate = new BindableNumber<double>();`
and bind that to `GameplayClockContainer.UserPlayback` instead?

Reply:
Yeah, something like that, if at all possible. Then you could even move the actual binding to `Player` by using `BindTarget` instead of passing it down.